### ISSUE TITLE
Why AMR with lambda_adv at 0 does not switch to VBPR?ASK]

### ISSUE THREAD

**Issue by xrob1** [PreferredAI/cornac#463]
Hi.
I am using AMR for a project in cornac. The first thing I started to do was to see if I can get VBPR results from AMR which I did by setting lambda_adv = 0. However, the results I am recieveing are far from VBPR (almost similar to AMR original value). Any idea how we could switch to VBPR by using AMR code?

amr = cornac.models.AMR(
    k=32,
    k2=32,
    n_epochs=1,
    batch_size=100,
    learning_rate=0.001,
    lambda_w=1,
    lambda_b=0.01,
    lambda_e=0.0,
    lambda_adv=0.0,
    use_gpu=True,
    name=f"AMR@{k}",
    seed=1498,
)

# Instantiate CVAE
vbpr = cornac.models.VBPR(
    k=32,
    k2=32,
    n_epochs=1,
    batch_size=100,
    learning_rate=0.001,
    lambda_w=1,
    lambda_b=0.01,
    lambda_e=0.0,
    use_gpu=True,
    name=f"VBPR",
    seed=1498,
)

       | NDCG@10 | Precision@10 | Recall@10 | Train (s) | Test (s)
------ + ------- + ------------ + --------- + --------- + --------
AMR@10 |  0.0041 |       0.0008 |    0.0071 |   24.5246 |   2.2947
VBPR   |  0.0213 |       0.0037 |    0.0312 |    1.3757 |   2.2822
BPR@10 |  0.0286 |       0.0059 |    0.0499 |    0.1687 |   2.1160

thank you for your help

**Comment by tqtg** [comment]
@RuihongQiu do you have any idea regarding this issue?

**Comment by tqtg** [comment]
@xrob1 implementations of the two models are not exactly the same. Therefore, you shouldn't expect the results to be the same when setting `lambda_adv=0`. That is what I can tell when looking at the implementation of the two models.
